src/Controllers/Version2/class-wc-rest-system-status-v2-controller.php

```diff
@@ -592,11 +588,30 @@ public function get_item_mappings() {
 	 * Get array of environment information. Includes thing like software
 	 * versions, and various server settings.
 	 *
+	 * @param array $fields List of fields to be included on the response.
 	 * @return array
 	 */
-	public function get_environment_info() {
+	public function get_environment_info( $fields ) {
 		global $wpdb;
 
+		$enable_remote_post = false;
+		$enable_remote_get  = false;
+		if ( in_array( 'environment', $fields, true ) ) {
+			$exclude = array();
+			foreach ( $fields as $field ) {
+				$values = explode( '.', $field );
+
+				if ( 'environment' !== $values[0] || empty( $values[1] ) ) {
+					continue;
+				}
+
+				$exclude[] = $values[1];
+			}
+
+			$enable_remote_post = 0 <= count( array_intersect( array( 'remote_post_successful', 'remote_post_response' ), $exclude ) );
+			$enable_remote_get  = 0 <= count( array_intersect( array( 'remote_get_successful', 'remote_get_response' ), $exclude ) );
+		}
```

Comment:
A nitpick, but I think it might be nicer to factor this out to a new function.

Reply:
Solved